secrets/secrets.go

```diff
@@ -128,6 +128,30 @@ func (s *Secrets) AddRegexRules(patterns []string) error {
 	return nil
 }
 
+func addCustomRules(rules []CustomRuleConfiguration) ([]Rule, error) {
+	var customRules []Rule
+	customRules = make([]Rule, len(rules))
+	for idx, rule := range rules {
+		regex, err := regexp.Compile(rule.RegexPattern)
+		if err != nil {
+			return nil, fmt.Errorf("failed to compile custom regex rule %s: %w", rule.RuleID, err)
+		}
+		customRules[idx] = Rule{
+			Rule: config.Rule{
+				Description: rule.Description,
+				RuleID:      rule.RuleID,
+				Regex:       regex,
+				Keywords:    []string{},
+			},
+			Tags: rule.Tags,
+		}
+		if rule.SecretGroup != 0 {
+			customRules[idx].Rule.SecretGroup = rule.SecretGroup
+		}
+	}
+	return customRules, nil
+}
+
```

Comment:
I was thinking about a simpler solution. Instead of building a collection of configurations to build a collection of custom rules from it, I think maybe we can follow the current implementation of taking rules from _gitleaks_, but if they do not exist, take them from our side.

What will it look like?

Imagine a folder `secrets/rules` here, and inside it a file for each rule, exactly like it in _gitleaks_, then consume some rules from here instead of there.

What do you think? Can you please check if it can work?

The other advantage here is that if it will be accepted by _gitleaks_, we will be able to easily move our rules there.

Reply:
I changed the implementation to match how rules are consumed from gitleaks